src/modules/dream/test/DreamBaseTest.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2021 European Spallation Source, see LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief test for class DreamBase
+///
+//===----------------------------------------------------------------------===//
+
+#include <string>
+
+#include <dream/DreamBase.h>
+#include <test/TestUDPServer.h>
+#include <test/SaveBuffer.h>
+#include <test/TestBase.h>
+#include <readout/ReadoutParser.h>
+
+class DreamBaseStandIn : public Jalousie::DreamBase {
+public:
+  DreamBaseStandIn(BaseSettings Settings, struct Jalousie::DreamSettings ReadoutSettings)
+      : Jalousie::DreamBase(Settings, ReadoutSettings){};
+  ~DreamBaseStandIn() = default;
+  using Detector::Threads;
+  using Jalousie::DreamBase::Counters;
+};
+
+class DreamBaseTest : public ::testing::Test {
+public:
+  void SetUp() override {
+    Settings.RxSocketBufferSize = 100000;
+    Settings.NoHwCheck = true;
+  }
+  void TearDown() override {}
+
+  std::chrono::duration<std::int64_t, std::milli> SleepTime{400};
+  BaseSettings Settings;
+  Jalousie::DreamSettings LocalSettings;
+};
+
+TEST_F(DreamBaseTest, Constructor) {
+  DreamBaseStandIn Readout(Settings, LocalSettings);
+  EXPECT_EQ(Readout.Counters.RxPackets, 0);
+}
+
+/// | ESS Header    |
+/// | Data header 1 | Readout 1 | Readout 2 | Readout 3 |
+/// | Data header 2 |
+/// | Data block 1  |
+/// | Data header 1 |
+/// | Data block 1  |
+///
+std::vector<uint8_t> TestPacket2{
+    // ESS header
+                0x00, 0x00, // pad, v0
+    0x45, 0x53, 0x53, 0x60, //  'E' 'S' 'S' 0x00
```

Comment:
Does the comment say the last byte should be 0x00?

Reply:
well spotted, will fix